// @emotion/babel-preset-css-prop/dist/babel-preset-css-prop.cjs.prod.js
"use strict";
function _interopDefault(ex) {
  return ex && "object" == typeof ex && "default" in ex ? ex.default : ex;
}
Object.defineProperty(exports, "__esModule", { value: !0 });
var _objectAssign = _interopDefault(require("object-assign")),
  jsx = _interopDefault(require("@babel/plugin-transform-react-jsx")),
  pragmatic = _interopDefault(require("babel-plugin-jsx-pragmatic")),
  emotion = _interopDefault(require("babel-plugin-emotion"));
function _objectWithoutPropertiesLoose(source, excluded) {
  if (null == source) return {};
  var key,
    i,
    target = {},
    sourceKeys = Object.keys(source);
  for (i = 0; i < sourceKeys.length; i++)
    (key = sourceKeys[i]),
      0 <= excluded.indexOf(key) || (target[key] = source[key]);
  return target;
}
var pragmaName = "___EmotionJSX",
  index = function(_ref) {
    _ref.pragma;
    var sourceMap = _ref.sourceMap,
      autoLabel = _ref.autoLabel,
      labelFormat = _ref.labelFormat,
      instances = _ref.instances,
      options = _objectWithoutPropertiesLoose(_ref, [
        "pragma",
        "sourceMap",
        "autoLabel",
        "labelFormat",
        "instances"
      ]);
    return {
      plugins: [
        [
          pragmatic,
          { export: "jsx", module: "@emotion/core", import: pragmaName }
        ],
        [
          jsx,
          _objectAssign(
            { pragma: pragmaName, pragmaFrag: "React.Fragment" },
            options
          )
        ],
        [
          emotion,
          {
            sourceMap: sourceMap,
            autoLabel: autoLabel,
            labelFormat: labelFormat,
            instances: instances,
            cssPropOptimization: !0
          }
        ]
      ]
    };
  };
exports.default = index;
